SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -53,16 +53,23 @@ Worker_EntityId SpatialVirtualWorkerTranslator::GetServerWorkerEntityForVirtualW
 
 void SpatialVirtualWorkerTranslator::ApplyVirtualWorkerManagerData(Schema_Object* ComponentObject)
 {
-	UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) ApplyVirtualWorkerManagerData"), *LocalPhysicalWorkerName);
+	UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("ApplyVirtualWorkerManagerData for %s:"), *LocalPhysicalWorkerName);
 
 	// The translation schema is a list of mappings, where each entry has a virtual and physical worker ID.
 	ApplyMappingFromSchema(ComponentObject);
 
-	for (const auto& Entry : VirtualToPhysicalWorkerMapping)
+	if (VirtualToPhysicalWorkerMapping.Num())
 	{
-		UE_LOG(LogSpatialVirtualWorkerTranslator, Verbose,
-			   TEXT("Translator assignment: Virtual Worker %d to %s with server worker entity: %lld"), Entry.Key, *(Entry.Value.WorkerName),
-			   Entry.Value.ServerWorkerEntityId);
+		if (LoadBalanceStrategy.IsValid() && LoadBalanceStrategy->IsReady())
+		{
+			UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("\t-> Strategy: %s"), *LoadBalanceStrategy->ToString());
```

Comment:
Little fuzzy what'll get stripped out when the UE_LOG macro is nulled, but might be worth wrapping this whole if statement in a NO_LOGGING check?